EasyCommands/BlockHandlers/TerminalBlockHandlers.cs

```diff
@@ -79,12 +79,12 @@ public class TerminalBlockHandler<T> : BlockHandler<T> where T : class, IMyTermi
                 AddPropertyHandler(ValueProperty.ACTION, new SimplePropertyHandler<T>((b, p) => p.attributeValue.GetValue(), (b, p, v) => b.ApplyAction(CastString(p.attributeValue.GetValue())), ResolvePrimitive(0)));
 
                 AddDirectionHandlers(Property.DIRECTION, Direction.FORWARD,
-                    TypeHandler(VectorHandler(b => Normalize(GetBlock2WorldTransform(b).Forward)), Direction.FORWARD),
-                    TypeHandler(VectorHandler(b => Normalize(GetBlock2WorldTransform(b).Backward)), Direction.BACKWARD),
-                    TypeHandler(VectorHandler(b => Normalize(GetBlock2WorldTransform(b).Up)), Direction.UP),
-                    TypeHandler(VectorHandler(b => Normalize(GetBlock2WorldTransform(b).Down)), Direction.DOWN),
-                    TypeHandler(VectorHandler(b => Normalize(GetBlock2WorldTransform(b).Left)), Direction.LEFT),
-                    TypeHandler(VectorHandler(b => Normalize(GetBlock2WorldTransform(b).Right)), Direction.RIGHT));
+                    TypeHandler(VectorHandler(b => b.WorldMatrix.Forward), Direction.FORWARD),
+                    TypeHandler(VectorHandler(b => b.WorldMatrix.Backward), Direction.BACKWARD),
+                    TypeHandler(VectorHandler(b => b.WorldMatrix.Up), Direction.UP),
+                    TypeHandler(VectorHandler(b => b.WorldMatrix.Down), Direction.DOWN),
+                    TypeHandler(VectorHandler(b => b.WorldMatrix.Left), Direction.LEFT),
+                    TypeHandler(VectorHandler(b => b.WorldMatrix.Right), Direction.RIGHT));
```

Comment:
Are these vectors still normalized?  Part of the intention was that all of the directional vectors had a magnitude of 1, to make scaling easy without having to manually normalize directions for every usage.

Reply:
`WorldMatrix` is constructed as normalized. You can check it by supplying non normalized vectors into `Matrix.CreateWorld` and checking the directions.